staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -196,6 +196,47 @@ func (c *CloudNodeLifecycleController) MonitorNodes(ctx context.Context) {
 	}
 }
 
+// getProviderID returns the provider ID for the node. If Node CR has no provider ID,
+// it will be the one from the cloud provider.
+func getProviderID(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node) (string, error) {
+	if instanceV2, ok := cloud.InstancesV2(); ok {
+		metadata, err := instanceV2.InstanceMetadata(ctx, node)
+		if err != nil {
+			if err == cloudprovider.InstanceNotFound {
+				return "", nil
+			}
+			return "", err
+		}
+
+		return metadata.ProviderID, nil
+	}
+
+	instances, ok := cloud.Instances()
+	if !ok {
+		return "", errors.New("cloud provider does not support instances")
+	}
+
+	providerID := node.Spec.ProviderID
+	if providerID == "" {
```

Comment:
One last nit: We should return early here

```suggestion
	if providerID != "" {
		return providerID, nil
	}
```

Reply:
Also regarding [Code Health: Reduce Nesting, Reduce Complexity](https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html)

Reply:
Good hints indeed, thank you, just fixed the code!